lib/api/src/conversions/vectors.rs

```diff
@@ -464,12 +464,11 @@ impl TryFrom<grpc::Vector> for VectorInternal {
         // sparse vector
         if let Some(indices) = vector.indices {
             return Ok(VectorInternal::Sparse(
-                sparse::common::sparse_vector::SparseVector::new(indices.data, vector.data)
```

Comment:
the correct error pattern was already present here, in that diff I just removed the unnecessary path :)